fast-list.js

```diff
@@ -257,6 +257,17 @@ FastList.prototype = {
       for (var j = endIndex; j >= startIndex; --j) renderItem(j);
     }
 
+    // When the data change we need to make sure we're not keeping
+    // outdated pre-rendered content lying around
+    if (reload) {
+      var fullLength = source.getFullLength();
+      for (var idx in items) {
+        if (idx >= fullLength) {
+          items[idx].dataset.populated = false;
+        }
+      }
+    }
```

Comment:
I'd prefer to see logic like this moved into separate functions to keep things tidy :) Not sure if that could impact performance slightly.